.github/workflows/test.yaml

```
@@ -0,0 +1,34 @@
+name: run tests
+
+on:
+  pull_request:
+    branches: [ "*" ]
+
+permissions:
+  id-token: write
+
+jobs:
+
+  testGo:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup Golang caches
+      uses: actions/cache@v4
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-golang-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-golang-
+
+    - name: Setup Golang
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.22'
+
+    - name: run go tests
+      run: |
+        go test -v github.com/adevinta/go-log-toolkit/...
```

No need to resolve it publicly if we trying to run the checked out repo, right?
```suggestion
        go test -v ./...
```